code/modules/mob/living/silicon/ai/ai.dm

```diff
@@ -228,9 +228,16 @@ var/list/ai_list = list()
 			var/area/borg_area
 			for(var/mob/living/silicon/robot/R in connected_robots)
 				borg_area = get_area(R)
+				var/robot_status = "Normal"
+				if(R.stat || !R.client)
+					robot_status = "OFFLINE"
+				else if(!R.cell)
+					robot_status = "DEPOWERED"
+				else if(R.cell.charge <= 0)

```

Comment:
You can combine this check with the above check
```else if(!R.cell || R.cell.charge <= 0)```
Since the right handside will only ever be evaluated if the cell exists.